Source/ACE.Server/Command/Handlers/DeveloperFixCommands.cs

```diff
@@ -803,68 +803,77 @@ public static void HandleVerifyExperience(Session session, params string[] param
+                        attributeXP += attribute.Value.CPSpent;
 
-                foreach (var vital in player.Biota.PropertiesAttribute2nd)
-                    vitalXP += vital.Value.CPSpent;
+                    foreach (var vital in player.Biota.PropertiesAttribute2nd)
+                        vitalXP += vital.Value.CPSpent;
 
-                foreach (var skill in player.Biota.PropertiesSkill)
-                    skillXP += skill.Value.PP;
+                    foreach (var skill in player.Biota.PropertiesSkill)
+                        skillXP += skill.Value.PP;
 
-                // find any xp spent on augs
-                var heritage = (HeritageGroup?)player.GetProperty(PropertyInt.HeritageGroup);
-                if (heritage == null)
-                    continue;       // ignore admins who have morphed into asheron / bael'zharon
+                    // find any xp spent on augs
+                    var heritage = (HeritageGroup?)player.GetProperty(PropertyInt.HeritageGroup);
+                    if (heritage == null)
+                        continue;       // ignore admins who have morphed into asheron / bael'zharon
 
-                var heritageAug = GetHeritageAug(heritage.Value);
+                    var heritageAug = GetHeritageAug(heritage.Value);
 
-                foreach (var kvp in AugmentationDevices)
-                {
-                    var augProperty = kvp.Key;
+                    foreach (var kvp in AugmentationDevices)
+                    {
+                        var augProperty = kvp.Key;
 
-                    var numAugs = player.GetProperty(augProperty) ?? 0;
-                    if (augProperty == heritageAug)
-                        numAugs--;
+                        var numAugs = player.GetProperty(augProperty) ?? 0;
+                        if (augProperty == heritageAug)
+                            numAugs--;
 
-                    if (numAugs <= 0)
-                        continue;
+                        if (numAugs <= 0)
+                            continue;
 
-                    var aug = DatabaseManager.World.GetCachedWeenie(kvp.Value);
-                    aug.PropertiesInt64.TryGetValue(PropertyInt64.AugmentationCost, out var costPer);
+                        var aug = DatabaseManager.World.GetCachedWeenie(kvp.Value);
+                        aug.PropertiesInt64.TryGetValue(PropertyInt64.AugmentationCost, out var costPer);
 
-                    augXP += costPer * numAugs;
-                }
+                        augXP += costPer * numAugs;
+                    }
 
-                var calculatedSpent = attributeXP + vitalXP + skillXP + augXP + diffXP;
+                    // Asheron's Lesser Benediction augmentation operates differently than all other augs.
+                    var hasLesserBenedictionAug = ctx.CharacterPropertiesQuestRegistry.Count(i => i.CharacterId == player.Guid.Full && i.QuestName.Equals("LesserBenedictionAug")) == 1;
```

Comment:
should use Any() at the minimum

looping through every record in the entire unindexed quest registry on the server, every time for each player, seems very unoptimized and slow. does this add any significant speed reduction to running this operation?

this should be building a lookup table beforehand

Reply:
```
            HashSet<uint> lesserBenediction = null;

            using (var ctx = new ShardDbContext())
            {
                // Asheron's Lesser Benediction augmentation operates differently than all other augs
                lesserBenediction = ctx.CharacterPropertiesQuestRegistry.Where(i => i.QuestName.Equals("LesserBenedictionAug")).Select(i => i.CharacterId).ToHashSet();
            }

                    if (lesserBenediction.Contains(player.Guid.Full))
                        augXP += 2000000000;
```

Reply:
sounds good to me